src/python/pants/goal/run_tracker.py

```diff
@@ -124,6 +124,9 @@ def __init__(self, *args, **kwargs):
     # Note that multiple threads may share a name (e.g., all the threads in a pool).
     self._threadlocal = threading.local()
 
+    # A logger facade that logs into this RunTracker.
+    self._logger = RunTrackerLogger(self)
```

Comment:
This has the RunTracker acting as middle-man so that folks who just need a logger now need to instead get a runtracker and pull on its logger. It seems a bit better to me to:
1. short term: prefer those folks construct a RunTrackerLogger for which they depend on a RunTracker subsystem
2. cleanup: have those folks depend on a Logger subsystem directly since it's what they actually need. That thin susbsystem would depend on a RunTracker (for now).

Reply:
We could do 2. straight away, I think. All uses of this logger today are via `Context.log`. But before I mess with that any further I'd like to understand how v2 tasks handle logging.